Automattic/newspack-plugin
# fix(perfmatters): use ignore-defaults constant when applying unused CSS

### All Submissions:






### Changes proposed in this Pull Request:



Fixes an issue with the handing of Perfmatters-ignore-defaults environment variable (`NEWSPACK_IGNORE_PERFMATTERS_DEFAULTS`). It was **not** handled when filtering the unused CSS excluded stylesheets.

This was introduced in https://github.com/Automattic/newspack-plugin/pull/2709

See 1200550061930446-as-1206352424304970/f

#### Migration

If a site had `NEWSPACK_IGNORE_PERFMATTERS_DEFAULTS` set, it's Perfmatters rules will have to be manually updated to ensure there will be no regressions. Any site with this env. variable set has to have the actual value of the `perfmatters_options` option updated to include the section (`rucss_excluded_stylesheets`) which was previously force-injected.  

To do that, the following should be done:

1. Save the intended value to a temporary file: `wp eval "echo wp_json_encode(\Newspack\Perfmatters::add_rucss_excluded_stylesheets(get_option('perfmatters_options')['assets']['rucss_excluded_stylesheets']));" | jq '.' > /tmp/permatters-rucss.json`
2. Update the nested option value with the value from the file: `wp option patch update perfmatters_options assets rucss_excluded_stylesheets --format=json < /tmp/permatters-rucss.json`

### How to test the changes in this Pull Request:

1. On `release`, install and activate Perfmatters plugin
2. Create an empty page with "One Column Wide" template 
3. Remove the `perfmatters_options` option, so there are no customisations present to the Perfmatters config
4. Add the `NEWSPACK_IGNORE_PERFMATTERS_DEFAULTS` environment variable (with `true` value)
5. Go to Settings -> Perfmatters -> Assets, and toggle on the "Remove Unused CSS" setting and save changes
6. View the empty page source: observe the `link#newspack-style-css` element has the `href` attribute, meaning the styles will be downloaded at page load _(this is the bug – Perfmatters should delay loading of this stylesheet because it's not present in the  "Excluded Stylesheets" setting)_
7. Switch to this branch, reload the page, observe that the `link#newspack-style-css` element has a `data-pmdelayedstyle` attribute instead of the `href` attribute – which means the download will be delayed by Perfmatters

### Other information:
